### DISCUSSION TITLE
scene object location

### DISCUSSION THREAD

**Discussion by mgood7123** [ezEngine/ezEngine#726] [Q&A] [answered]
```
scene
> player - mesh sphere - script component
> enemy - mesh sphere
```

how would i get `player` to be able to locate `enemy` ?

for example

```
player

get scene
    -> find by name "enemy"
        -> [object] get location
            -> do something with returned location
```

for example

```
player

get scene
    -> find by name "wall"
        -> stop movement if player location intersects wall
```

**Comment by jankrassnigg** [comment, marked as answer]
There are different ways, depending on your use case.

1. The easiest one is to set a [Global Key](https://ezengine.net/pages/docs/runtime/world/game-objects.html#global-keys) on an object, then you can query for that one particular object through ezWorld.
2. A more powerful approach is to use the [spatial system](https://ezengine.net/pages/docs/runtime/world/spatial-system.html) to query for objects with certain properties in a particular area.
3. Another method is to use the physics system to query for physically simulated objects in an area (usually a sphere overlap test). This is useful if you know that the desired objects are physics objects anyway.

To start, my recommended way would be to attach a [marker component](https://ezengine.net/pages/docs/gameplay/marker-component.html) to your enemy, set the marker to "enemy" or so, and then use the spatial system to find all enemies close to your player.

These queries usually give you GameObjects. Then you can use [TryGetComponent](https://ezengine.net/pages/docs/runtime/world/components.html#querying-components-from-game-objects) on that object to check whether it has the desired components.

Note that the Visual Scripting system isn't very powerful, and you can't do these checks there. The TypeScript integration has way more functionality exposed, so you can do spatial system and physics queries there. However, I would actually recommend to use a [C++ plugin](https://ezengine.net/pages/docs/custom-code/cpp/cpp-project-generation.html) since that gives you the full feature set, and the workflow is actually pretty good by now.

**Reply by mgood7123** [reply]
alright, ill try typescript :)

**Reply by jankrassnigg** [reply]
Note that the "Testing Chambers" project uses several TS scripts to implement its game logic. There are also area queries in there (for example in the BallMine code and the Turret). This is even covered in [this video](https://www.youtube.com/watch?v=T1eDenHWPKc).
